e3db-crypto-android/src/main/java/com/tozny/e3db/AndroidCrypto.java

```diff
@@ -20,54 +20,82 @@
 
 package com.tozny.e3db;
 
+import com.goterl.lazycode.lazysodium.LazySodium;
+import com.goterl.lazycode.lazysodium.LazySodiumAndroid;
+import com.goterl.lazycode.lazysodium.exceptions.SodiumException;
+import com.goterl.lazycode.lazysodium.interfaces.Box;
+import com.goterl.lazycode.lazysodium.interfaces.SecretBox;
+import com.goterl.lazycode.lazysodium.interfaces.SecretStream;
+import com.goterl.lazycode.lazysodium.interfaces.Sign;
 import com.tozny.e3db.crypto.*;
-import org.libsodium.jni.NaCl;
-import org.libsodium.jni.Sodium;
-import org.libsodium.jni.crypto.Box;
-import org.libsodium.jni.crypto.Random;
-import org.libsodium.jni.crypto.SecretBox;
+
 import static com.tozny.e3db.Checks.*;
-import static org.libsodium.jni.Sodium.*;
+import com.goterl.lazycode.lazysodium.SodiumAndroid;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
 
 class AndroidCrypto implements Crypto {
-  private final Sodium sodium;
-  private final Random random;
+  private static final int BLOCK_SIZE = 65_536; // 2 ^ 16
```

Comment:
This is a rather large block size. What is this constant used for?

Reply:
That is the size of each block that we encrypt with SecretStream. The size of the block affects two things:

1) Number of crypto operations (encrypt/decrypt each block)
2) Additional space required by the encrypted file (each block holds an authentication tag).

The size increase is negligible (20bytes per block I think). The number of processing steps is more of a concern. I started with 4192 sized blocks, but that produces 244K blocks on a 1GB file. Since we have to store at least two of the blocks in memory during encryption, I didn't want to pick a block size that was too big (e.g., 1MB). So I picked this as a reasonable size.